src/buffer.jl

```diff
@@ -0,0 +1,432 @@
+# Support for python buffer protocol buffers
+
+# for versions > 3.2 (TODO: obj field not in struct for python versions <= 3.2)
```

Comment:
@jakebolewski, why do you say that the `obj` field is not in the struct for earlier versions?

I just looked in the Python 2.7.6 `Include/object.h` header file, and `Py_buffer` is defined as
```c
/* Py3k buffer interface */
typedef struct bufferinfo {
    void *buf;
    PyObject *obj;        /* owned reference */
    Py_ssize_t len;
    Py_ssize_t itemsize;  /* This is Py_ssize_t so it can be                    
                             pointed to by strides in simple case.*/
    int readonly;
    int ndim;
    char *format;
    Py_ssize_t *shape;
    Py_ssize_t *strides;
    Py_ssize_t *suboffsets;
    Py_ssize_t smalltable[2];  /* static store for shape and strides of         
                                  mono-dimensional buffers. */
    void *internal;
} Py_buffer;
```
which seems identical except for the `smalltable` field (which is missing in Python 3.4).  However, we can just add some padding to the structure for compatibility, since we don't actually access those fields ourselves.